.changeset/mighty-olives-wash.md

```diff
@@ -0,0 +1,5 @@
+---
+'@astrojs/compiler': patch
+---
+
+Fix an issue where when there are nested expressions, subsequent content was incorrectly treated as plain text in some cases.
```

Comment:
```suggestion
Fixes an issue where when there are nested expressions, subsequent content was incorrectly treated as plain text in some cases.
```

Reply:
Noted for the future, thanks @sarah11918 🙌